ops/c/src/externlib/ops_hdf5.cpp

```diff
@@ -1506,3 +1585,325 @@ void ops_write_const_hdf5(char const *name, int dim, char const *type,
 
   H5Fclose(file_id);
 }
+
+hsize_t calc_total_plane_size(const ops_dat &data,
+                              const int cross_section_dir) {
+  const int space_dim{data->block->dims};
+  int *size{new int(space_dim)};
+  for (int d = 0; d < space_dim; d++) {
+    size[d] = data->size[d] - (data->d_p[d] - data->d_m[d]);
+  }
+
+  hsize_t number_element{data->elem_size};
+
+  if (cross_section_dir == 0) {
```

Comment:
Please make this whole function more generic to any dimensions with loops. Also insert check that cross_section_dir < space_dim

Reply:
Sorted now.